client/lib/cart-values/cart-items.js

```diff
@@ -404,7 +410,11 @@ function themeItem( themeSlug, source ) {
  * @returns {Object} the new item as `CartItemValue` object
  */
 function domainRegistration( properties ) {
-	return assign( domainItem( properties.productSlug, properties.domain, properties.source ), { is_domain_registration: true } );
+	return assign( domainItem( properties.productSlug, properties.domain, properties.source ),
+		{
+			is_domain_registration: true,
+			...( properties.extra ? { extra: properties.extra } : {} ),
```

Comment:
The tests added here won't pass without it, because the extra field is stripped in domainItem. 
I think this is ok as is, or we can add "native" support for 'extra' to domainItem - whatever you think is better.